tcms/testplans/views.py

```diff
@@ -361,69 +361,76 @@ def get(request, plan_id, slug=None, template_name='plan/get.html'):
 
 @require_http_methods(['GET', 'POST'])
 @permission_required('testruns.change_testrun')
-def choose_run(request, plan_id):
+def choose_run(request, plan_id):  # pylint: disable=inconsistent-return-statements
```

Comment:
This requires a bit of non-straight forward thinking. The reason for this warning is that you have 2 if statement which could return a response and then you have the last line of the function body, which by default is equivalent to "return None". The easiest way to fix that is to simply add this explicit return at the end.

However you have a decorator which indicates that only GET and POST methods are accepted and the second if statement is really the ELSE part for the first if statement.

On top of that the two branches of this function don't have anything in common so you can split it out into 2 separate functions that will each handle the respective http request method.